servers/xr/xr_controller_tracker.h

```diff
@@ -0,0 +1,52 @@
+/**************************************************************************/
+/*  xr_controller_tracker.h                                               */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef XR_CONTROLLER_TRACKER_H
+#define XR_CONTROLLER_TRACKER_H
+
+#include "core/os/thread_safe.h"
+#include "servers/xr/xr_positional_tracker.h"
+
+/**
+	The controller tracker object as an object that represents the position and orientation of a controller.
+*/
+
+class XRControllerTracker : public XRPositionalTracker {
+	GDCLASS(XRControllerTracker, XRPositionalTracker);
+	_THREAD_SAFE_CLASS_
+
+protected:
+	static void _bind_methods();
```

Comment:
Maybe for a follow up PR and as part of 4.4, but I think we should move the input functions on `XRPositionalTracker` here as they really only apply to controllers.

We can introduce compatibility functions on `XRPositionalTracker` that check if the object is in fact a `XRControllerTracker` and provide the same functionality.

Another thing we should consider is whether we paint ourselves into a corner with the name "controller" as this will also be applicable for data gloves, and hand tracking if something like the hand tracking interaction profile is used.

This could even be a puck tracker like the HTC ones. Those are a bit of a weird case though. In their basic forms they are just positional trackers, but you can use the included USB connector to attach a device to them and turn them into full controllers (like a fully functional and tracked gun prop). 

This was why in the past we used a pretty ambiguous name for tracking. 
Though this may simply be something that we explain in documentation, that controller should be seen in the widest form, any tracked device that provides further inputs and outputs such as buttons, triggers and haptics.

Reply:
I marked the methods in XRPositionalTracker as deprecated and indicated XRControllerTracker should be used.

I also have the input functions perform checks to emit deprecated warnings if they're invoked on a non XRControllerTracker instance via:
```cpp
if (!dynamic_cast<const XRControllerTracker *>(this)) {
    WARN_DEPRECATED_MSG(R"*(The "get_input()" method is deprecated, use "XRControllerTracker" instead.)*");
}
```

For now due to the methods being inherited that's about the best I can do for deprecation warnings/annotations.